File: zs-leetcode-go/BalancedBinaryTree/BalancedBinaryTree.go
package BalancedBinaryTree

import "for-an-offer/zs-leetcode-go/types"

func aux(root *types.TreeNode, res *bool) int {
	if root == nil {
		return 0
	}
	left := aux(root.Left, res)+1
	right := aux(root.Right, res)+1
	dep := left-right
	if !(dep >= -1 && dep <= 1){
		*res = false
		return 0
	}
	if dep == -1 {
		return right
	}
	return left
}
func isBalanced(root *types.TreeNode) bool {
	res := true
	aux(root, &res)
	return res
}

File: sjq-leetcode-C++/balanced-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool flag = true;
    bool isBalanced(TreeNode* root) {
        if(root == NULL) return true;
        int left = depth(root);
        return flag;
    }
    int depth(TreeNode* root){
        if(root == NULL) return 0;
        int left = depth(root->left);
        int right = depth(root->right);
        if(abs(left-right)>1) flag=false;
        return max(left, right)+1;
    }
};
